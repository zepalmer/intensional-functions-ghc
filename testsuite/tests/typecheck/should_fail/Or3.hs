{-# LANGUAGE OrPatterns #-}
{-# LANGUAGE GADTs #-}

module Main where

data G a where
  G1 :: Num a => G a
  G2 :: Num a => G a
  G3 :: Num a => G a

bar :: G a -> a
bar (one of G2, G1) = 3

data GADT a where
    IsInt1 :: GADT Int
    IsInt2 :: GADT Int

foo :: a -> GADT a -> a
foo x (one of IsInt1 {}, IsInt2 {}) = x + 1

f x = case x of
  (one of Left a, Right a) -> a

g x = case x of
  (one of _, (one of _, x)) -> x

main = print $ foo 3 IsInt1