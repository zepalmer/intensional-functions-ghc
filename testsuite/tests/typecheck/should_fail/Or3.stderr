
Or3.hs:12:6: error: [GHC-81303]
    An or-pattern may not bind (type) variables nor type class or equality constraints: one of G2,
                                                                                               G1

Or3.hs:19:8: error: [GHC-81303]
    An or-pattern may not bind (type) variables nor type class or equality constraints: one of IsInt1 {},
                                                                                               IsInt2 {}

Or3.hs:22:4: error: [GHC-81303]
    An or-pattern may not bind variables: one of Left a, Right a

Or3.hs:25:15: error: [GHC-81303]
    An or-pattern may not bind variables: one of _, x
